crates/lib/src/transaction/validator.rs

```diff
@@ -239,63 +247,149 @@ impl TransactionValidator {
 }
 
 pub async fn validate_token_payment(
-    rpc_client: &RpcClient,
     transaction: &Transaction,
-    validation: &ValidationConfig,
     required_lamports: u64,
+    validation: &ValidationConfig,
+    rpc_client: &RpcClient,
     _signer_pubkey: Pubkey,
 ) -> Result<(), KoraError> {
     let mut total_lamport_value = 0;
-    let token_program = TokenProgram::new(TokenType::Spl);
 
-    for ix in transaction.message.instructions.iter() {
-        if *ix.program_id(&transaction.message.account_keys) != token_program.program_id() {
-            continue;
-        }
+    // Check both SPL and Token2022 transfers
+    for token_type in [TokenType::Spl, TokenType::Token2022] {
```

Comment:
you shouldn't need to iterate twice. You can check the instruction program id against these two program ids to see which one to use on line 263.